Cilsil/Services/CfgParserService.cs

```diff
@@ -88,38 +92,47 @@ private void ComputeMethodCfg(MethodDefinition method)
 
             // True if the translation terminates early, false otherwise.
             var translationUnfinished = false;
-            if (!method.IsAbstract && methodBody.Instructions.Count > 0)
+            foreach (var exceptionHandler in methodBody.ExceptionHandlers)
             {
-                programState.PushInstruction(methodBody.Instructions.First());
-                do
+                var exceptionHandlingBlock = exceptionHandler;
+                var exceptionHandlingBlockStartOffset = -1;
+                var exceptionHandlingBlockEndOffset = -1;
+                TypeReference catchType = null;
+                switch (exceptionHandlingBlock.HandlerType) 
                 {
-                    var nextInstruction = programState.PopInstruction();
-                    // Checks if there is a node for the offset that we can reuse.
-                    (var nodeAtOffset, var excessiveVisits) =
-                        programState.GetOffsetNode(nextInstruction.Offset);
-                    if (nodeAtOffset != null)
-                    {
-                        programState.PreviousNode.Successors.Add(nodeAtOffset);
-                    }
-                    else if (excessiveVisits)
-                    {
-                        TimeoutMethodCount++;
-                        Log.WriteError("Translation timeout.");
-                        Log.RecordUnfinishedMethod(programState.Method.GetCompatibleFullName(),
-                                                   nextInstruction.RemainingInstructionCount());
-                        translationUnfinished = true;
+                    case ExceptionHandlerType.Catch:
+                        catchType = exceptionHandlingBlock.CatchType;
+                        exceptionHandlingBlockStartOffset = exceptionHandlingBlock.HandlerStart.Offset;
+                        exceptionHandlingBlockEndOffset = exceptionHandlingBlock.HandlerEnd.Offset;
                         break;
-                    }
-                    else if (!InstructionParser.ParseCilInstruction(nextInstruction, programState))
-                    {
-                        Log.RecordUnfinishedMethod(programState.Method.GetCompatibleFullName(),
-                                                   nextInstruction.RemainingInstructionCount());
-                        translationUnfinished = true;
+
+                    case ExceptionHandlerType.Finally:
+                        exceptionHandlingBlockStartOffset = exceptionHandlingBlock.HandlerStart.Offset;
+                        exceptionHandlingBlockEndOffset = exceptionHandlingBlock.HandlerStart.Offset;
                         break;
+
+                    case ExceptionHandlerType.Filter:
```

Comment:
are these handler types not associated with an exception handling block? add a short comment explaining what's going on in these cases

Reply:
Done